azurelinuxagent/common/utils/archive.py

```diff
@@ -164,6 +168,12 @@ class StateArchiver(object):
     def __init__(self, lib_dir):
         self._source = os.path.join(lib_dir, 'history')
 
+        if not os.path.isdir(self._source):
+            try:
+                fileutil.mkdir(self._source, mode=0o700)
+            except IOError as e:
+                self.logger.error(u"Failed to create archive history dir: {0}", e)
```

Comment:
Just log `e.strerror`, which is the description text for the actual error. I don't think you need the entire exception. Also, log the name of the directory (self._source) that couldn't be created.